### PR TITLE
DOC: RT03 fix for various DataFrameGroupBy and SeriesGroupBy methods

### PR DESCRIPTION
**Pull request by quangngd**
- [x] xref https://github.com/pandas-dev/pandas/issues/57416
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -240,6 +240,7 @@ def apply(self, func, *args, **kwargs) -> Series:
         Returns
         -------
         Series or DataFrame
+            The resulting dtype will reflect the return value of the passed ``func``.
```
Comment (datapythonista) [inline pandas/core/groupby/generic.py:243]: This doesn't seem very clear to me. Why not something like `A pandas object with the result of applying ``func`` to each group.`. Or maybe you can think of something better, but the return description should focus on what's being returned, not which type is being returned.

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -1550,6 +1552,7 @@ def apply(self, func, *args, include_groups: bool = True, **kwargs) -> NDFrameT:
         Returns
         -------
         Series or DataFrame
+            The resulting dtype will reflect the return value of the passed ``func``.
```
Comment (datapythonista) [inline pandas/core/groupby/groupby.py:1555]: Same as above.

**Review by datapythonista — COMMENTED** [review]
Thanks for working on this @quangngd.

Very nice job, there is just one description that is not very clear, at least to me, see the comment.

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (quangngd) [reply pandas/core/groupby/generic.py]: Make sense. Couldn't think of anything else so I use yours.

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (quangngd) [reply pandas/core/groupby/groupby.py]: Updated

File: pandas/core/groupby/groupby.py [hunk 3]
```diff
@@ -3511,11 +3515,12 @@ def resample(self, rule, *args, include_groups: bool = True, **kwargs) -> Resamp
...
-        pandas.api.typing.DatetimeIndexResamplerGroupby,
-        pandas.api.typing.PeriodIndexResamplerGroupby, or
-        pandas.api.typing.TimedeltaIndexResamplerGroupby
+        pandas.api.typing.DatetimeIndexResampler
             Return a new groupby object, with type depending on the data
-            being resampled.
+            being resampled:
+            - pandas.api.typing.DatetimeIndexResamplerGroupby
+            - pandas.api.typing.PeriodIndexResamplerGroupby
+            - pandas.api.typing.TimedeltaIndexResamplerGroupby
```
Comment (datapythonista) [inline pandas/core/groupby/groupby.py:3523]: This doesn't seem to be rendering correctly: https://pandas.pydata.org/preview/pandas-dev/pandas/57862/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.resample.html#pandas.core.groupby.DataFrameGroupBy.resample

Also, the type is incorrect now. Why don't you leave as type simply: `DatetimeIndexResampler, PeriodIndexResampler or TimdeltaResampler`

And in the description you can have something like `Resampler object for the type of the index`.

Comment (quangngd) [reply pandas/core/groupby/groupby.py]: Updated. But i'll hold this up and wait for the new flow regarding alll this doc-related thingy

**Comment by datapythonista** [comment]
@quangngd we changed how we specify the pending errors in `code_checks.sh` and you'll have to resolve the conflicts. Sorry for the inconvenience.

**Comment by quangngd** [comment]
@datapythonista resolved

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @quangngd

**Pull request merged**